Failed to upgrade rook cluster from 1.1.9 to custom version(1.2.3)

<!-- **Are you in the right place?**
1. For issues or feature requests, please create an issue in this repository.
2. For general technical and non-technical questions, we are happy to help you on our [Rook.io Slack](https://slack.rook.io/).
3. Did you already search the existing open issues for anything similar? -->

**Is this a bug report or feature request?**
* Bug Report
I have build a custom image to test upgrade from 1.1.9 to 1.2.3(which is next rook release) the operator pod is crashing due to ceph version check failure in recomcile

**Deviation from expected behavior:**

**Expected behavior:**
Upgrade should be successful

**How to reproduce it (minimal and precise):**
<!-- Please let us know any circumstances for reproduction of your bug. -->
* Deploy rook using v1.1.9 branch
* Create a custom image from release-v1.2 branch and upgrade rook
**File(s) to submit**:

* Cluster CR (custom resource), typically called `cluster.yaml`, if necessary
* Operator's logs, if necessary
* Crashing pod(s) logs, if necessary

 To get logs, use `kubectl -n <namespace> logs <pod name>`
When pasting logs, always surround them with backticks or use the `insert code` button from the Github UI.
Read [Github documentation if you need help](https://help.github.com/en/articles/creating-and-highlighting-code-blocks).

**Environment**:
* OS (e.g. from /etc/os-release):
* Kernel (e.g. `uname -a`):
* Cloud provider or hardware configuration:
* Rook version (use `rook version` inside of a Rook Pod): 1.1.9
* Storage backend version (e.g. for ceph do `ceph -v`):
* Kubernetes version (use `kubectl version`): 1.17.1
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift):
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):


operator logs:
```
020-01-29 06:17:44.985016 I | operator: Watching all namespaces for cluster CRDs
2020-01-29 06:17:44.985029 I | op-cluster: start watching clusters in all namespaces
2020-01-29 06:17:44.985070 I | op-cluster: Enabling hotplug orchestration: ROOK_DISABLE_DEVICE_HOTPLUG=false
I0129 06:17:44.985864       8 leaderelection.go:217] attempting to acquire leader lease  rook-ceph/rook.io-block...
2020-01-29 06:17:44.986455 I | operator: setting up the controller-runtime manager
2020-01-29 06:17:45.023114 I | op-cluster: starting cluster in namespace rook-ceph
2020-01-29 06:17:45.503978 I | operator: starting the controller-runtime manager
E0129 06:17:45.726018       8 runtime.go:69] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:76
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:65
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:51
/usr/local/go/src/runtime/panic.go:522
/usr/local/go/src/runtime/panic.go:82
/usr/local/go/src/runtime/signal_unix.go:390
/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:211
/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:157
/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:74
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153
/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88
/usr/local/go/src/runtime/asm_amd64.s:1337
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x165a342]
goroutine 472 [running]:
github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:58 +0x105
panic(0x1bd05e0, 0x35ca600)
	/usr/local/go/src/runtime/panic.go:522 +0x1b5
github.com/rook/rook/pkg/operator/ceph/cluster/crash.getImageVersion(0x19eb1ab, 0xb, 0xc000cdda71, 0xf, 0xc000808840, 0x9, 0x0, 0x0, 0xc000808860, 0x9, ...)
	/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:211 +0x82
github.com/rook/rook/pkg/operator/ceph/cluster/crash.(*ReconcileNode).reconcile(0xc000b01760, 0x0, 0x0, 0xc000a4bb48, 0x6, 0xc00079fcb0, 0x60ec4cf5e0, 0xc02b2f674e, 0x2b2f674e00414237)
	/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:157 +0x704
github.com/rook/rook/pkg/operator/ceph/cluster/crash.(*ReconcileNode).Reconcile(0xc000b01760, 0x0, 0x0, 0xc000a4bb48, 0x6, 0x35ed680, 0xc000a90900, 0xc00079fd88, 0xc00079fdb8)
	/home/mrajanna/workspace/src/github.com/rook/rook/pkg/operator/ceph/cluster/crash/reconcile.go:74 +0x69
github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler(0xc0003ac1e0, 0x1c692c0, 0xc0008eadc0, 0x1c69200)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:216 +0x146
github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem(0xc0003ac1e0, 0xc0000ff200)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:192 +0xb5
github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).worker(0xc0003ac1e0)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:171 +0x2b
github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000469390)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x54
github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000469390, 0x3b9aca00, 0x0, 0x1, 0xc0009d2f60)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc000469390, 0x3b9aca00, 0xc0009d2f60)
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start
	/home/mrajanna/workspace/src/github.com/rook/rook/vendor/sigs.k8s.io/controller-runtime/pkg/internal/controller/controller.go:157 +0x311
```

ceph cluster crd

```
kuberc get cephcluster -o yaml
apiVersion: v1
items:
- apiVersion: ceph.rook.io/v1
  kind: CephCluster
  metadata:
    creationTimestamp: "2020-01-29T05:42:03Z"
    finalizers:
    - cephcluster.ceph.rook.io
    generation: 6
    name: rook-ceph
    namespace: rook-ceph
    resourceVersion: "4208"
    selfLink: /apis/ceph.rook.io/v1/namespaces/rook-ceph/cephclusters/rook-ceph
    uid: 42f847bb-f048-4e9b-88a7-009668ce85ff
  spec:
    cephVersion:
      allowUnsupported: true
      image: ceph/ceph:v14.2.4-20190917
    dashboard:
      enabled: true
      ssl: true
    dataDirHostPath: /var/lib/rook
    disruptionManagement: {}
    external:
      enable: false
    mgr:
      modules:
      - enabled: true
        name: pg_autoscaler
    mon:
      allowMultiplePerNode: true
      count: 1
    monitoring:
      rulesNamespace: rook-ceph
    network:
      hostNetwork: false
      provider: ""
      selectors: null
    rbdMirroring:
      workers: 0
    removeOSDsIfOutAndSafeToRemove: false
    storage:
      config:
        databaseSizeMB: "1024"
        journalSizeMB: "1024"
        osdsPerDevice: "1"
      directories:
      - config: null
        path: /var/lib/rook
      storageClassDeviceSets: null
      useAllDevices: false
      useAllNodes: true
  status:
    ceph:
      health: HEALTH_OK
      lastChecked: "2020-01-29T06:16:04Z"
    state: Created
kind: List
metadata:
  resourceVersion: ""
  selfLink: ""
```

--- a/pkg/operator/ceph/cluster/crash/reconcile.go
+++ b/pkg/operator/ceph/cluster/crash/reconcile.go
@@ -208,7 +208,7 @@ func getImageVersion(cephCluster cephv1.CephCluster) (*version.CephVersion, erro
 	for i := 0; i < getVersionMaxRetries; i++ {
 		// If the Ceph cluster has not yet recorded the image and version for the current image in its spec, then the Crash
 		// controller should wait for the version to be detected.
-		if cephCluster.Spec.CephVersion.Image == cephCluster.Status.CephVersion.Image {
+		if cephCluster.Status.CephVersion != nil && cephCluster.Spec.CephVersion.Image == cephCluster.Status.CephVersion.Image {
 			logger.Debugf("ceph version found %q", cephCluster.Status.CephVersion.Version)
 			return spec.ExtractCephVersionFromLabel(cephCluster.Status.CephVersion.Version)
 		}

